app/services/supplementary-billing/abstraction-billing-period.service.js

```diff
@@ -137,23 +139,61 @@ function _abstractionPeriods (billingPeriod, chargePurpose) {
 }
 
 /**
- * Calculates the 'billable days' for each abstraction billing period
+ * Calculates the 'billable days' for each abstraction billable period
  *
  * @param {Object[]} abstractionPeriods An array of abstraction billing periods
  *
  * @returns {Object[]} The array abstraction periods each with a new `billableDays` property
  * @returns {Date} periods[].startDate
  * @returns {Date} periods[].endDate
  * @returns {boolean} periods[].consider
+ * @returns {Date} periods[].billableStartDate
+ * @returns {Date} periods[].billableEndDate
  * @returns {number} periods[].billableDays
  */
 function _calculateBillableDays (abstractionPeriods) {
-  let difference
-  let billableDays
   for (const abstractionPeriod of abstractionPeriods) {
-    difference = abstractionPeriod.endDate.getTime() - abstractionPeriod.startDate.getTime() // difference in msecs
-    billableDays = Math.ceil(difference / (1000 * 3600 * 24)) + 1 // (1000 msecs * (60 secs * 60 mins) * 24 hrs)
-    abstractionPeriod.billableDays = billableDays
+    if (abstractionPeriod.billableStartDate) {
+      const difference = abstractionPeriod.billableEndDate.getTime() - abstractionPeriod.billableStartDate.getTime() // difference in msecs
+      const billableDays = Math.ceil(difference / (1000 * 3600 * 24)) + 1 // (1000 msecs * (60 secs * 60 mins) * 24 hrs)
+      abstractionPeriod.billableDays = billableDays
+    }
+  }
+}
+
+/**
+ * Calculates the 'billable period' for each abstraction billing period
+ *
+ * @param {Object[]} abstractionPeriods An array of abstraction billing periods
+ * @param {Object} billingPeriod Object that has a `startDate` and `endDate` that defines the billing period
+ *
+ * @returns {Object[]} The array abstraction periods each with new `billableStartDate` & `billableEndDate` properties
+ * @returns {Date} periods[].startDate
+ * @returns {Date} periods[].endDate
+ * @returns {boolean} periods[].consider
+ * @returns {Date} periods[].billableStartDate
+ * @returns {Date} periods[].billableEndDate
```

Comment:
Is it worth clarifying that `billableStartDate` and `billableEndDate` aren't always set? Only if the start date is on or before the end date? (Same applies for L146 I guess, where we don't always set the `billableDays` property, just if `billableStartDate` is defined)

Reply:
See my previous comment. We should just scrub `@returns`. Much less confusing.

Reply:
I'll get rid of the `@returns`